src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -235,24 +238,74 @@ public function verifyEmail($verificationNonce)
         $secondFactorToVerify->verifyEmail();
     }
 
-    public function vetSecondFactor($registrationCode, $secondFactorIdentifier, $documentNumber, $identityVerified)
-    {
-        foreach ($this->verifiedSecondFactors as $secondFactor) {
-            if (!$secondFactor->wouldBeVettedBy(
-                $registrationCode,
-                $secondFactorIdentifier,
-                $documentNumber,
-                $identityVerified
-            )) {
-                continue;
-            }
+    public function vetSecondFactor(
+        IdentityApi $identity,
+        SecondFactorId $secondFactorId,
+        $secondFactorIdentifier,
+        $registrationCode,
+        $documentNumber,
+        $identityVerified
+    ) {
+        /** @var VettedSecondFactor|false $secondFactorWithHighestLoa */
+        $secondFactorWithHighestLoa = array_reduce(
+            $this->vettedSecondFactors->toArray(),
+            function (VettedSecondFactor $carry, VettedSecondFactor $item) {
+                return $carry->hasEqualOrHigherLoaComparedTo($item) ? $carry : $item;
+            },
+            $this->vettedSecondFactors->first()
+        );
```

Comment:
What would you think about moving this into a custom `SecondFactorCollection` that extends `ArrayCollection` or something. Seems it might be a bit cleaner? Fairly complex code to read... just a thought.